openpathsampling/snapshot.py

```diff
@@ -313,22 +258,137 @@ def inner(self, *args, **kwargs):
 
     return _has
 
+@lazy_loading_attributes('_reversed')
+class AbstractSnapshot(StorableObject):
+    """
+    Simulation snapshot. Contains references to a configuration and momentum
+    """
+
+    def __init__(self, is_reversed=False, reversed_copy=None, topology=None):
```

Comment:
If all subclasses of `AbstractSnapshot` require an `engine` parameter in their `__init__` functions, shouldn't `AbstractSnapshot` require it too?